Code/Tools/ProjectManager/Source/ProjectsScreen.cpp

```diff
@@ -301,6 +301,11 @@ namespace O3DE::ProjectManager
                 }
             });
 
+
+            // It's more efficient to update the project engine by loading engine infos once
+            // instead of loading them all each time we want to know what project an engine uses
+            auto engineInfoResult = PythonBindingsInterface::Get()->GetAllEngineInfos();
```

Comment:
How long does this take if have 10 engines registered?

We don't want it to be unresponsive for the user opening the project dialog the first time.

I use 10 engines registered, because I don't think a user would any more than 10 engines that are actually found on the machine and not stale entries.
So that should be good cap on long it would take.

Reply:
Let me pull the numbers - I made this change because without it the Project Manager was much slower because it was inefficient in how it was asking for project data.  I really want us to move all our `PythonBindings` calls to the `project-manager-interface.py` where we can safely implement more efficient use of o3de python code and even limited caching.  I think we may need to address this so the UI doesn't freeze up if it tries to display 30+ projects and 1k+ gems that it knows about because of remote repos.  Some of the pages that will pull in remote content will need to use async updates, pagination at some point.

Reply:
With 12 projects and 9 engines this whole function (ResetProjectContents) takes ~500ms before these changes and ~150ms after based on 2-3 runs in each implementations.

This specific line of code `GetAllEngineInfos` takes ~65ms to load and parse 9 `engine.json`s found in the `o3de_manifest.json` so not really noticeable unless every time you get the project info you are loading every `engine.json` which is what was happening before.  Still lots of room for improvement in startup time - this function is run 3 times alone on startup!

Reply:
That is good enough and won't be noticeable by the user, if we do it on startup.